src/Bridge/Symfony/Routing/ApiLoader.php

```diff
@@ -99,16 +99,26 @@ private function addRoute(RouteCollection $routeCollection, string $resourceClas
             throw new RuntimeException('Either a "route_name" or a "method" operation attribute must exist.');
         }
 
-        if (isset($operation['controller'])) {
-            $controller = $operation['controller'];
-        } else {
-            $actionName = sprintf('%s_%s', strtolower($operation['method']), $collection ? 'collection' : 'item');
+        $collectionString = $collection ? 'collection' : 'item';
+        $controller = $operation['controller'] ?? null;
+
+        if (null === $controller) {
+            $actionName = sprintf('%s_%s', strtolower($operation['method']), $collectionString);
             $controller = self::DEFAULT_ACTION_PATTERN.$actionName;
         }
 
-        $path = '/'.$normalizedShortName;
-        if (!$collection) {
-            $path .= '/{id}';
+        if ($operationName !== strtolower($operation['method'])) {
```

Comment:
```php
/**
 * @Resource(itemOperations={
 *     "my_get_op"={"method"="GET"},
 * })
 */
```

will produce: `api_platform.action.my_get_op_item` which is not what we want.

After giving this more thought, I think we should be asking ourselves what is the intended / expected use case for custom operations... That should inform our naming scheme. Actually, there is no harm in using the old way. I don't think there should be more than one custom operations which use the same default action.